### PR TITLE
cache/redis: feature redis sentinel

### PR DESCRIPTION
**Pull request by bpereto**
##### SUMMARY

Add redis sentinel functionality to redis cache plugin. Used for high available redis setups controlled by redis sentinel.   
The plugin asks the sentinels which redis node is currently the active master and sets the connection internally. There is no change in logic how the redis cache backend works, only where to connect.

Redis Sentinel is available since redis python >=2.9.0 (https://github.com/andymccurdy/redis-py/commit/46e475d9b95501a46c7abba9f05d415e27e9a62d). the minimal requirement at the moment for the redis cache module is >= 2.4.5
To not break backwards compatibility, Sentinel is imported when a sentinel uri is configured. additionally the sentinel config is sepparated by `;` to maintain the existing uri layout.

merge #1036 first. this PR is built uppon.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

redis

##### ADDITIONAL INFORMATION

Example ansible.cfg:
```
[defaults]
fact_caching=redis
fact_caching_connection=localhost:26379;localhost:26380;0:changeme
fact_caching_redis_sentinel=mymaster
```

```paste below

```

### CODE REVIEW

**Comment by felixfontein** [comment]
I've marked it as WIP to make sure this isn't merged before the other redis cache PR has been merged. Once that has been merged, this can be rebased and the WIP removed.

**Comment by felixfontein** [comment]
This now needs to be rebased.

File: plugins/cache/redis.py
Comment (felixfontein) [inline plugins/cache/redis.py:130]: Can this potentially dump sensitive information (like password)?

**Review by felixfontein — COMMENTED** [review]
BTW, if this is no longer a WIP, please remove `[WIP]` from the PR's name.

File: plugins/cache/redis.py
Comment (Andersson007) [inline plugins/cache/redis.py:21]:
```suggestion
          - To use redis sentinel, use sepparator C(;), for example C(localhost:26379;localhost:26379;0:changeme). Requires redis>=2.9.0.
```

File: plugins/cache/redis.py
Comment (Andersson007) [inline plugins/cache/redis.py:46]:
```suggestion
        description: The redis sentinel service name (or referenced as cluster name).
```

File: plugins/cache/redis.py
Comment (Andersson007) [inline plugins/cache/redis.py:161]:
```suggestion
            raise AnsibleError('Could not connect to redis sentinel: %s' % to_native(exc))
```

**Comment by Andersson007** [comment]
`# (c) 2014, Brian Coca, Josh Drake, et al`
@bcoca @joshdrake

File: plugins/cache/redis.py
Comment (felixfontein) [reply plugins/cache/redis.py:21]: While we're at it: separator is written with one p, not with two.

File: plugins/cache/redis.py
Comment (bpereto) [reply plugins/cache/redis.py:130]: unlikely, the output, with configured password
```
Redis connection: Redis<ConnectionPool<Connection<host=localhost,port=6379,db=0>>>
```

Comment (bpereto) [reply plugins/cache/redis.py:130]: Output of configured redis with password:
```
Redis connection: Redis<SentinelConnectionPool<service=mymaster(master)>
```

What could leak is the Sentinel debug output on line 156
```
Using redis sentinel: [('localhost', '26379'), ('localhost', '26380')] with {'db': '0', 'password': 'changeme'}
```

Comment (felixfontein) [reply plugins/cache/redis.py:131]: In that case, definitely remove `kw` from that `display.vv` call in line 157.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@bpereto thanks for implementing this!
@Andersson007 thanks for reviewing!

**Comment by felixfontein** [comment]
@bpereto are you interested in becoming a maintainer for the redis cache plugin? That would mean you would get CC'ed by the bot for every PR and issue created for it, and your vote would count more on decisions whether something should be changed/added/...

**Pull request merged**
